Add task to check and set public videos

chriso123 opened the pull request:
Solution for https://github.com/llttugraz/moodle-oermod_opencast/issues/6

chriso123 committed:
WIP work on task to check video visibility

rearrange modules class, separated many function to api_helper to be
used by other classes to.
add message provider to send important errors to administrators
add task (WIP) to update video visibility if necessary

chriso123 committed:
Work on task and notification

The task now works through several steps:
* Load all opencast snapshots (without duplicates)
* Get the information through the opencast api
* Fix opencast role permissions (add anonymous, remove write from
  lecturer)
* Write error log and send notification for missing videos or other
  errors that lead to a non-functional video

chriso123 committed:
Fix some minor errors

chriso123 committed:
Add first unit test for api_helper

Use tool_opencast\local\api_testable for fake webservice data.
Add json files for webservice fake answers

chriso123 committed:
Add unit test stubs for all methods

chriso123 committed:
Add all tests for api_helper

chriso123 committed:
add test for message class

chriso123 committed:
add log check to message test

chriso123 committed:
add test for hook

move some code from api_helper_test to testcourse helper to reduce code
duplication.

chriso123 committed:
add unit test for check release task

add namespaces to unit test files

chriso123 committed:
add tests for module class

code coverage for all classes is now >90%

chriso123 committed:
resolve codechecker issues

chriso123 committed:
update github workflow

remove php < 8.2
add php 8.4
remove moodle < 4.5
add moodle 5.0 and 5.1

chriso123 committed:
change minimum Moodle version required

chriso123 committed:
increase postgres version

chriso123 committed:
add tool and block opencast to workflow

chriso123 committed:
fix select query for postgres

chriso123 committed:
add fix for unit test for postgres

chriso123 committed:
fix unit test for postgres

The last fix did not work, it called the same code that checks for
a database transaction again. So the solution is to disable database
transactions for the two tests that are sending emails

chriso123 committed:
set release version to 4.5 only

remove matrix settings in github workflow for Moodle 5.0 and 5.1. Before
this will be set in this subplugin, the base plugin has to be made ready
for 5.0+

chriso123 committed:
add message type failed

when the update for the video in opencast fails, also a message should
be send. So that this problem can be investigated manually.

chriso123 committed:
add coverage information for get_video_for_message

chriso123 committed:
ignore space after function -async (- in js

chriso123 committed:
increase version

chriso123 committed:
add php 8.1 to workflow, update readme

chriso123 committed:
update workflow

set ubuntu 24.04
set actions to v4

chriso123 committed:
fix sql

sql threw warning that identifier has been found multiple times.

chriso123 committed:
fix some sql issues, add term to description

To identify videos faster on opencast side, the term oer_published; will
be added to the videos description

chriso123 committed:
update sql in task to work with postgres

taraghi commented on tests/helper/testcourse.php:
```diff
@@ -0,0 +1,174 @@
+namespace oermod_opencast;
+
+use local_oer\identifier;
+use PHPUnit\Runner\Exception;
```
I assume, you meant to use `\Exception` instead, didn't you?

chriso123 committed:
remove phpunit exception import

chriso123 committed:
use subjects instead of description for oer tag

Instead of writing oer_published into the description, the subjects
metadata field is used. This field is also an array which makes it
easier to compare and add the data.

chriso123 committed:
fix copy&paste error

chriso123 committed:
remove checkbox on block_opencast view for oer

When a video has been released as oer, the teacher should not be able to
delete it through the block_opencast view.

chriso123 committed:
Add hook to add parameters to opencast studio url

chriso123 committed:
change to a reporting task

Instead of fixing issues the task will report the errors.
- A new setting for an email address
- Metadata subject is no longer written
- Send email to set email (main admin als fallback) and users with
  missingvideos capability

chriso123 committed:
fix unit tests, change how email is send

it is not possible to overwrite the email of an existing user directly
before using message_send, so email_to_user is used instead

chriso123 committed:
add licence check

update message to use a mustache template
add/update tests

chriso123 committed:
update covers tag

chriso123 committed:
fix german language string

chriso123 committed:
fix check for role setting

chriso123 committed:
fix codechecker

chriso123 committed:
Update README and CHANGES
